From 3bf2efe225e3737871fee01cdc23a84b8c5ba50f Mon Sep 17 00:00:00 2001
From: sudotac <sudo@tofuyard.net>
Date: Sun, 4 Oct 2026 21:21:16 +0900
Subject: [PATCH] timer_create0[13]: replace _sigev_un._tid with
 sigev_notify_thread_id

These tests use _sigev_un._tid member in struct sigevent. This member is
glibcism.

Use sigev_notify_thread_id instead for better portability. This makes it
possible to build these tests in musl-based systems.

Signed-off-by: sudotac <sudo@tofuyard.net>
---
 testcases/kernel/syscalls/timer_create/timer_create01.c | 2 +-
 testcases/kernel/syscalls/timer_create/timer_create03.c | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/testcases/kernel/syscalls/timer_create/timer_create01.c b/testcases/kernel/syscalls/timer_create/timer_create01.c
index a9f5d1bf2..dde6bca87 100644
--- a/testcases/kernel/syscalls/timer_create/timer_create01.c
+++ b/testcases/kernel/syscalls/timer_create/timer_create01.c
@@ -63,7 +63,7 @@ static void run(unsigned int n)
 			continue;
 
 		if (nt->sigev_notify == SIGEV_THREAD_ID)
-			evp._sigev_un._tid = getpid();
+			evp.sigev_notify_thread_id = getpid();
 
 		TEST(tst_syscall(__NR_timer_create, clock,
 				nt->sigev_notify ? &evp : NULL,
diff --git a/testcases/kernel/syscalls/timer_create/timer_create03.c b/testcases/kernel/syscalls/timer_create/timer_create03.c
index 309319296..fbc2abf72 100644
--- a/testcases/kernel/syscalls/timer_create/timer_create03.c
+++ b/testcases/kernel/syscalls/timer_create/timer_create03.c
@@ -37,7 +37,7 @@ static void run(void)
 
 	evp.sigev_signo  = SIGALRM;
 	evp.sigev_notify = SIGEV_SIGNAL | RANDOM_UNUSED_NUMBER;
-	evp._sigev_un._tid = getpid();
+	evp.sigev_notify_thread_id = getpid();
 
 	TEST(tst_syscall(__NR_timer_create, clock, &evp, &created_timer_id));
 
-- 
2.56.0

